Zend/zend_vm_def.h

```diff
@@ -4139,23 +4147,25 @@ ZEND_VM_HOT_HANDLER(131, ZEND_DO_FCALL_BY_NAME, ANY, ANY, SPEC(RETVAL,OBSERVER))
 		EG(current_execute_data) = execute_data;
 
 ZEND_VM_C_LABEL(fcall_by_name_end):
-		zend_vm_stack_free_args(call);
-
-		uint32_t call_info = ZEND_CALL_INFO(call);
-		if (UNEXPECTED(call_info & (ZEND_CALL_HAS_EXTRA_NAMED_PARAMS|ZEND_CALL_ALLOCATED))) {
-			if (call_info & ZEND_CALL_HAS_EXTRA_NAMED_PARAMS) {
-				zend_free_extra_named_params(call->extra_named_params);
-			}
-			zend_vm_stack_free_call_frame_ex(call_info, call);
-		} else {
-			EG(vm_stack_top) = (zval*)call;
-		}
 
 		if (!RETURN_VALUE_USED(opline)) {
 			i_zval_ptr_dtor(ret);
 		}
 	}
 
+ZEND_VM_C_LABEL(out):
```

Comment:
This has a subtle behavioral change in that it releases the return value before the arguments, whereas before it cleaned arguments first. I'm not sure if that matters. Alternatively, you could re-use the same label and just initialize `ret`. That said, it might be easy to miss in the future that the assumption of `call->func` being internal breaks from that label forward.

Reply:
Unfortunately it has also broken `Zend/tests/gh10072.phpt`. I've now effectively used your suggestion, but with moving the logic into an `if(0)` branch.